@Test public void testConvolve(){
  double[] x1={1.2,-1.8,1.4};
  double[] h1={1,0.8,0.5,0.3};
  double[] y1={1.2,-0.84,0.56,0.58,0.16,0.42};
  double tolerance=1e-13;
  double[] yActual=MathArrays.convolve(x1,h1);
  Assert.assertArrayEquals(y1,yActual,tolerance);
  double[] x2={1,2,3};
  double[] h2={0,1,0.5};
  double[] y2={0,1,2.5,4,1.5};
  yActual=MathArrays.convolve(x2,h2);
  Assert.assertArrayEquals(y2,yActual,tolerance);
  try {
    MathArrays.convolve(new double[]{1,2},null);
    Assert.fail("an exception should have been thrown");
  }
 catch (  NullArgumentException e) {
  }
  try {
    MathArrays.convolve(null,new double[]{1,2});
    Assert.fail("an exception should have been thrown");
  }
 catch (  NullArgumentException e) {
  }
  try {
    MathArrays.convolve(new double[]{1,2},new double[]{});
    Assert.fail("an exception should have been thrown");
  }
 catch (  NoDataException e) {
  }
  try {
    MathArrays.convolve(new double[]{},new double[]{1,2});
    Assert.fail("an exception should have been thrown");
  }
 catch (  NoDataException e) {
  }
  try {
    MathArrays.convolve(new double[]{},new double[]{});
    Assert.fail("an exception should have been thrown");
  }
 catch (  NoDataException e) {
  }
}
